src/hooks/billing/useTenantMissingPaymentMethodWarning.tsx

```diff
@@ -37,8 +37,10 @@ function useTenantMissingPaymentMethodWarning() {
         notificationStoreSelectors.hideNotification
     );
 
-    const { tenants } = useTenantDetails();
-    const hasSupportRole = useEntitiesStore_hasSupportRole();
+    const { tenantBillingDetails } = useTenantBillingDetails();
+    const hasSupportRole = useUserInfoSummaryStore(
+        (state) => state.hasSupportAccess
+    );
```

Comment:
This is the only thing that _actually_ needs to get stuff from the tenant table. So it is the main consumer here.

This is a bit overkill to check the support role here again... but to keep things aligning more 1 to 1 I just left it.